### PR TITLE
feat: include data action

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

https://www.notion.so/dust-tt/Tool-Include-c6f4310246b44c93bd61e1fe7ef6dff4?pvs=4

Basically, rename the "Most Recent Data" action into "Include Data". The timeframe is made optional, and by default disabled.

Screenshot 2024-11-08 at 17 51 48

Screenshot 2024-11-08 at 17 52 16

Screenshot 2024-11-08 at 17 52 29

## Risk

Well tested, but on assistant builder critical path and touches code that could affect Retrieval and Process on top of Most Recent.

## Deploy Plan

Deploy front

### CODE REVIEW

File: front/components/assistant_builder/actions/RetrievalAction.tsx [hunk 1]
```diff
@@ -85,18 +88,19 @@ export function hasErrorActionRetrievalExhaustive(
 ): string | null {
   return action.type === "RETRIEVAL_EXHAUSTIVE" &&
     Object.keys(action.configuration.dataSourceConfigurations).length > 0 &&
-    !!action.configuration.timeFrame.value
+    // The time frame is optional for exhaustive retrieval, but if it is set, it must be valid.
```
Comment (Fraggle) [inline front/components/assistant_builder/actions/RetrievalAction.tsx:91]: thanks for the comment

File: front/components/assistant_builder/submitAssistantBuilderForm.ts [hunk 2]
```diff
@@ -70,13 +89,7 @@ export async function submitAssistantBuilderForm({
             name: a.name,
             description: a.description,
             query: a.type === "RETRIEVAL_SEARCH" ? "auto" : "none",
-            relativeTimeFrame:
-              a.type === "RETRIEVAL_EXHAUSTIVE"
-                ? {
-                    duration: a.configuration.timeFrame.value,
-                    unit: a.configuration.timeFrame.unit,
-                  }
-                : "auto",
```
Comment (Fraggle) [inline front/components/assistant_builder/submitAssistantBuilderForm.ts:79]: what is "auto" that has been removed ?

**Review by Fraggle — APPROVED** [review]
LGTM.

File: front/components/assistant_builder/submitAssistantBuilderForm.ts (hunk 2 above)
Comment (fontanierh) [reply front/components/assistant_builder/submitAssistantBuilderForm.ts:79]: hasn't been removed (see above line 65). It's just what we use for regular retrieval

Comment (fontanierh) [reply front/components/assistant_builder/submitAssistantBuilderForm.ts:79]: (i.e the model decides the time frame)

File: front/lib/api/assistant/actions/names.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ export const DEFAULT_BROWSE_ACTION_NAME = "browse";
 export const DEFAULT_PROCESS_ACTION_NAME =
   "extract_structured_data_from_data_sources";
 export const DEFAULT_RETRIEVAL_ACTION_NAME = "search_data_sources";
-// The name below is used by the assistant builder when the user selects "Most recent data".
+// The name below is used by the assistant builder when the user selects "Include data".
 export const DEFAULT_RETRIEVAL_NO_QUERY_ACTION_NAME =
```
Comment (spolu) [inline front/lib/api/assistant/actions/names.ts:7]: Should we rename the action name presented to the model?

Comment (spolu) [reply front/lib/api/assistant/actions/names.ts:7]: Like cleared + we'll need the inclue name once we start using the action for JIT data source interactions cc @Fraggle

**Comment by spolu** [comment]
Nice job pushing this out!!

**Pull request merged**